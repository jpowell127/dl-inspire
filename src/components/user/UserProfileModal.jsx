import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import UserWorkExperience from "./UserWorkExperience";
import UserProjects from "./UserProjects";
import UserStats from "./UserStats";
import UserFollow from "./UserFollow";
import { Button, Header, Modal, Image, Grid } from "semantic-ui-react";

export default function UserProfileModal({ user }) {
  const {
    id,
    display_name,
    username,
    occupation,
    company,
    location,
    stats
  } = user;
  const image = user.images[100];

  const button = (
    <Button fluid inverted color="pink">
      View User
    </Button>
  );

  return (
    <Modal closeIcon trigger={button}>
      <Header as="h1" size="huge">
        <Image spaced="right" floated="right" rounded src={image} />
        {display_name} (aka {username})
        <Header.Subheader size="medium">
          {company || occupation}
        </Header.Subheader>
        <Header.Subheader size="medium">{location}</Header.Subheader>
      </Header>
      <Header as="h1" size="huge">
        <UserStats stats={stats} />
      </Header>
      <Grid columns={3} centered padded divided>
        <Grid.Row>
          <Grid.Column>
            <UserProjects userId={id} />
          </Grid.Column>
          <Grid.Column>
            <UserWorkExperience username={username} />
          </Grid.Column>
          <Grid.Column>
            <UserFollow username={username} />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Modal>
  );
}

UserProfileModal.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    display_name: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    occupation: PropTypes.string,
    company: PropTypes.string,
    location: PropTypes.string,
    images: PropTypes.object.isRequired,
    stats: PropTypes.shape({
      followers: PropTypes.number.isRequired,
      following: PropTypes.number.isRequired,
      appreciations: PropTypes.number.isRequired,
      views: PropTypes.number.isRequired,
      comments: PropTypes.number.isRequired
    }).isRequired
  }).isRequired
};
