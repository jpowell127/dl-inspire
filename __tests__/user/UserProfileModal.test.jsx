import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import UserProfileModal from "../../src/components/user/UserProfileModal";

test("UserProfileModal renders correctly", () => {
  const user = {
    id: 1,
    display_name: "test-user",
    username: "testuser",
    occupation: "designer",
    company: "test co",
    location: "somewhere",
    images: {},
    stats: {
      followers: 1,
      following: 1,
      appreciations: 1,
      views: 1,
      comments: 1
    }
  };
  const component = shallow(<UserProfileModal user={user} />);
  expect(component).toMatchSnapshot();
});
