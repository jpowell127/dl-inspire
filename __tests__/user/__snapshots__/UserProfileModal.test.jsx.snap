// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileModal renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfileModal
    user={
      Object {
        "company": "test co",
        "display_name": "test-user",
        "id": 1,
        "images": Object {},
        "location": "somewhere",
        "occupation": "designer",
        "stats": Object {
          "appreciations": 1,
          "comments": 1,
          "followers": 1,
          "following": 1,
          "views": 1,
        },
        "username": "testuser",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
          size="huge"
        >
          <Image
            as="img"
            floated="right"
            rounded={true}
            spaced="right"
            src={undefined}
            ui={true}
          />
          test-user
           (aka 
          testuser
          )
          <HeaderSubheader
            size="medium"
          >
            test co
          </HeaderSubheader>
          <HeaderSubheader
            size="medium"
          >
            somewhere
          </HeaderSubheader>
        </Header>,
        <Header
          as="h1"
          size="huge"
        >
          <UserStats
            stats={
              Object {
                "appreciations": 1,
                "comments": 1,
                "followers": 1,
                "following": 1,
                "views": 1,
              }
            }
          />
        </Header>,
        <Grid
          centered={true}
          columns={3}
          divided={true}
          padded={true}
        >
          <GridRow>
            <GridColumn>
              <UserProjects
                userId={1}
              />
            </GridColumn>
            <GridColumn>
              <UserWorkExperience
                username="testuser"
              />
            </GridColumn>
            <GridColumn>
              <UserFollow
                username="testuser"
              />
            </GridColumn>
          </GridRow>
        </Grid>,
      ],
      "closeIcon": true,
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "trigger": <Button
        as="button"
        color="pink"
        fluid={true}
        inverted={true}
        role="button"
      >
        View User
      </Button>,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": Array [
            <Image
              as="img"
              floated="right"
              rounded={true}
              spaced="right"
              src={undefined}
              ui={true}
            />,
            "test-user",
            " (aka ",
            "testuser",
            ")",
            <HeaderSubheader
              size="medium"
            >
              test co
            </HeaderSubheader>,
            <HeaderSubheader
              size="medium"
            >
              somewhere
            </HeaderSubheader>,
          ],
          "size": "huge",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "img",
              "floated": "right",
              "rounded": true,
              "spaced": "right",
              "src": undefined,
              "ui": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "test-user",
          " (aka ",
          "testuser",
          ")",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test co",
              "size": "medium",
            },
            "ref": null,
            "rendered": "test co",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "somewhere",
              "size": "medium",
            },
            "ref": null,
            "rendered": "somewhere",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": <UserStats
            stats={
              Object {
                "appreciations": 1,
                "comments": 1,
                "followers": 1,
                "following": 1,
                "views": 1,
              }
            }
          />,
          "size": "huge",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "stats": Object {
              "appreciations": 1,
              "comments": 1,
              "followers": 1,
              "following": 1,
              "views": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centered": true,
          "children": <GridRow>
            <GridColumn>
              <UserProjects
                userId={1}
              />
            </GridColumn>
            <GridColumn>
              <UserWorkExperience
                username="testuser"
              />
            </GridColumn>
            <GridColumn>
              <UserFollow
                username="testuser"
              />
            </GridColumn>
          </GridRow>,
          "columns": 3,
          "divided": true,
          "padded": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn>
                <UserProjects
                  userId={1}
                />
              </GridColumn>,
              <GridColumn>
                <UserWorkExperience
                  username="testuser"
                />
              </GridColumn>,
              <GridColumn>
                <UserFollow
                  username="testuser"
                />
              </GridColumn>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <UserProjects
                  userId={1}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "userId": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <UserWorkExperience
                  username="testuser"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "username": "testuser",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <UserFollow
                  username="testuser"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "username": "testuser",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
            size="huge"
          >
            <Image
              as="img"
              floated="right"
              rounded={true}
              spaced="right"
              src={undefined}
              ui={true}
            />
            test-user
             (aka 
            testuser
            )
            <HeaderSubheader
              size="medium"
            >
              test co
            </HeaderSubheader>
            <HeaderSubheader
              size="medium"
            >
              somewhere
            </HeaderSubheader>
          </Header>,
          <Header
            as="h1"
            size="huge"
          >
            <UserStats
              stats={
                Object {
                  "appreciations": 1,
                  "comments": 1,
                  "followers": 1,
                  "following": 1,
                  "views": 1,
                }
              }
            />
          </Header>,
          <Grid
            centered={true}
            columns={3}
            divided={true}
            padded={true}
          >
            <GridRow>
              <GridColumn>
                <UserProjects
                  userId={1}
                />
              </GridColumn>
              <GridColumn>
                <UserWorkExperience
                  username="testuser"
                />
              </GridColumn>
              <GridColumn>
                <UserFollow
                  username="testuser"
                />
              </GridColumn>
            </GridRow>
          </Grid>,
        ],
        "closeIcon": true,
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "trigger": <Button
          as="button"
          color="pink"
          fluid={true}
          inverted={true}
          role="button"
        >
          View User
        </Button>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": Array [
              <Image
                as="img"
                floated="right"
                rounded={true}
                spaced="right"
                src={undefined}
                ui={true}
              />,
              "test-user",
              " (aka ",
              "testuser",
              ")",
              <HeaderSubheader
                size="medium"
              >
                test co
              </HeaderSubheader>,
              <HeaderSubheader
                size="medium"
              >
                somewhere
              </HeaderSubheader>,
            ],
            "size": "huge",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "img",
                "floated": "right",
                "rounded": true,
                "spaced": "right",
                "src": undefined,
                "ui": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "test-user",
            " (aka ",
            "testuser",
            ")",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "test co",
                "size": "medium",
              },
              "ref": null,
              "rendered": "test co",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "somewhere",
                "size": "medium",
              },
              "ref": null,
              "rendered": "somewhere",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": <UserStats
              stats={
                Object {
                  "appreciations": 1,
                  "comments": 1,
                  "followers": 1,
                  "following": 1,
                  "views": 1,
                }
              }
            />,
            "size": "huge",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "stats": Object {
                "appreciations": 1,
                "comments": 1,
                "followers": 1,
                "following": 1,
                "views": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "centered": true,
            "children": <GridRow>
              <GridColumn>
                <UserProjects
                  userId={1}
                />
              </GridColumn>
              <GridColumn>
                <UserWorkExperience
                  username="testuser"
                />
              </GridColumn>
              <GridColumn>
                <UserFollow
                  username="testuser"
                />
              </GridColumn>
            </GridRow>,
            "columns": 3,
            "divided": true,
            "padded": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridColumn>
                  <UserProjects
                    userId={1}
                  />
                </GridColumn>,
                <GridColumn>
                  <UserWorkExperience
                    username="testuser"
                  />
                </GridColumn>,
                <GridColumn>
                  <UserFollow
                    username="testuser"
                  />
                </GridColumn>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <UserProjects
                    userId={1}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "userId": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <UserWorkExperience
                    username="testuser"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "username": "testuser",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <UserFollow
                    username="testuser"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "username": "testuser",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
